cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -163,7 +163,8 @@ func (scaleSet *ScaleSet) Autoprovisioned() bool {
 func (scaleSet *ScaleSet) GetOptions(defaults config.NodeGroupAutoscalingOptions) (*config.NodeGroupAutoscalingOptions, error) {
 	template, err := scaleSet.getVMSSFromCache()
 	if err != nil {
-		return nil, err
+		klog.Errorf("failed to get information for VMSS: %s", scaleSet.Name)
```

Comment:
`GetOptions` shouldn't error. Failure to get info for VMSS should result in defaulting instead. Erroring here results in extra logs from core that we don't need.

Reply:
Is this true of `GetOptions` in general? In which chase we can remove the error from the return. Otherwise, let's add a comment in code to explain why the error is not returned.

Reply:
I linked some more details about this in the PR description, but we have GetOptions() interface-level function definition: https://github.com/kubernetes/autoscaler/blob/9b41b9dc5bf431a4ad657c044e809b15cc28a7e2/cluster-autoscaler/cloudprovider/cloud_provider.go#L241-L244

**If options aren't found (e.g. == nil), then core will use default options.**

---
### AWS + GCE implementations
```
// AWS
func (ng *AwsNodeGroup) GetOptions(defaults config.NodeGroupAutoscalingOptions) (*config.NodeGroupAutoscalingOptions, error) {
	if ng.asg == nil || ng.asg.Tags == nil || len(ng.asg.Tags) == 0 {
		return &defaults, nil
	}
	return ng.awsManager.GetAsgOptions(*ng.asg, defaults), nil
}

// GCE
func (mig *gceMig) GetOptions(defaults config.NodeGroupAutoscalingOptions) (*config.NodeGroupAutoscalingOptions, error) {
	return mig.gceManager.GetMigOptions(mig, defaults), nil
}
```
---
### Without this change 
If instead we error due to vmss not found, then we will get more `vmss not found` logs than we need to. For the sake of moving this bug fix forward, I'm happy to revert this part and return to it later.

**Callers of GetOptions**
Refs from deletion/scale down:
  * static autoscaler: uses default if err + opts == nil; no panic, **errors** 
  * group deletion scheduler: **fails if err != nil**, sets empty if opts == nil, **err not used**
  * budgets: **err logged**
  * scaledown planner: **err logged**
  * scaledown set processor: **errors**

Refs from scaleup:
  * orchestrator scaleup:  **err not used**
  * node_group_config_processor: **errors**

Reply:
No need to revert, let's just add a comment in the code (rather than PR comment) to explain why the error is not returned.